Document APIResponseCompression feature gate (#41318)
kubernetes/website: content/en/docs/reference/using-api/api-concepts.md

<followed by regular watch stream starting from resourceVersion="10245">
```

## Response compression

{{< feature-state for_k8s_version="v1.16" state="beta" >}}

`APIResponseCompression` is an option that allows the API server to compress the responses for **get**
and **list** requests, reducing the network bandwidth and improving the performance of large-scale clusters.

> It is enabled by default since Kubernetes 1.16, but can be disabled by setting the `--feature-gates`
> flag on the API server to `APIResponseCompression=false`.

Reviewer: ```suggestion
It is enabled by default since Kubernetes 1.16 and it can be disabled by including
`APIResponseCompression=false` in the `--feature-gates` flag on the API server.
```

> but can be disabled by setting the `--feature-gates` flag on the API server to `APIResponseCompression=false`.

Reviewer: This paragraph smells like a note about the "feature gate" rather than the feature.
The gates may come and go, but we are more focused on the feature that comes with it.
Suppose the feature gate is GA'ed and/or eventually removed, can we simply remove a few sentences rather than rewriting the paragraph?

Author: Done. 

I left _can be disabled by setting the `--feature-gates` flag on the API server to `APIResponseCompression=false`_ because it's one of the asks in #30639:

> Can a client turn it off? Can a client demand a compressed response?

> `APIResponseCompression` can significantly reduce the size of the response, especially for large resources or

Reviewer: ```suggestion
API response compression can significantly reduce the size of the response, especially for large resources or
```

Author: Done

[collections](/docs/reference/using-api/api-concepts/#collections).
For example, a **list** request for pods can return hundreds of kilobytes or even megabytes of data,
depending on the number of pods and their attributes. By compressing the response, the network bandwidth
can be saved and the latency can be reduced.

> To verify if `APIResponseCompression` is working, you can send a **get** or **list** request to the

Reviewer 2: nit: if it has a request header, it's a GET (**list** is a logical verb in API and is not part of HTTP)
